.github/workflows/coordinator-test.yml

```diff
@@ -1,31 +1,120 @@
 name: Coordinator Tests
 
+# TODO: add paths for pull requests and pushes
+# run on:
+# - manual trigger
 on:
   workflow_dispatch:
 
+# global env vars, available in all jobs and steps
+env:
+  ARTIFACTORY_USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
+  ARTIFACTORY_PASSWORD: ${{ secrets.ARTIFACTORY_PASSWORD }}
+  MAVEN_OPTS: '-Xmx4g'
+
 jobs:
+
+  unit-test:
+    name: Unit tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - uses: actions/setup-java@v3
+        name: Setup Java JDK
+        with:
+          distribution: 'temurin'
+          java-version: 8
+          cache: 'maven'
+
+      - name: Setup Maven
+        run: |
+          mkdir -p ~/.m2
+          cat <<EOF > ~/.m2/settings.xml
+          <settings>
+            <servers>
+              <server>
+                <id>artifactory-snapshots</id>
+                <username>${ARTIFACTORY_USERNAME}</username>
+                <password>${ARTIFACTORY_PASSWORD}</password>
+              </server>
+              <server>
+                <id>artifactory-releases</id>
+                <username>${ARTIFACTORY_USERNAME}</username>
+                <password>${ARTIFACTORY_PASSWORD}</password>
+             </server>
+           </servers>
+          </settings>
+          EOF
+
+      # TODO: get rid of the CCM in the unit tests
+      #  all unit tests should run without storage
+      - name: Install CCM
+        run: |
+          python -m pip install --upgrade pip ccm
+          ccm list
+
+      - name: Run unit tests
+        run: |
+          ./mvnw -B -ntp clean test -P dse
+
   integration-test:
-    name: Integration Test
-    # CCM installation trickier on 20.04, alas; Python 2.x dep etc problematic
-    runs-on: ubuntu-18.04
+    name: Integration tests
+    runs-on:
+      labels: stargate-coord-runner
```

Comment:
speed improvements with 2 vs 8 core github runner

```
c 4
 * 2 cores: 54m47s
 * 8 cores: 50m43s

c 3.11
 * 2 cores: 51m31s  
 * 8 cores: 52m18s

dse
 * 2 cores: 1h24m
 * 8 cores: 1h10m 
```

I don't think there is any significant difference here